server/src/main/java/org/opensearch/tasks/TaskThreadContextStatePropagator.java

```diff
@@ -20,7 +20,8 @@
  * Propagates TASK_ID across thread contexts
  */
 public class TaskThreadContextStatePropagator implements ThreadContextStatePropagator {
-    @Override
+
+    @SuppressWarnings("removal")
```

Comment:
Why we are suppressing warnings here? And `@Override` is fine as well, right?

Reply:
it's giving the compilation failure 

```
error: warnings found and -Werror specified
/var/jenkins/workspace/gradle-check/search/server/src/main/java/org/opensearch/telemetry/tracing/ThreadContextBasedTracerContextStorage.java:75: warning: [removal] headers(Map<String,Object>) in ThreadContextStatePropagator has been deprecated and marked for removal
    public Map<String, String> headers(Map<String, Object> source) {
                               ^
/var/jenkins/workspace/gradle-check/search/server/src/main/java/org/opensearch/telemetry/tracing/ThreadContextBasedTracerContextStorage.java:54: warning: [removal] transients(Map<String,Object>) in ThreadContextStatePropagator has been deprecated and marked for removal
    public Map<String, Object> transients(Map<String, Object> source) {
```

Reply:
Ah I see, right, I think it comes with JDK-21, thank you

Reply:
Yes, that's an accidental change as I was trying to get rid of the warning. Adding it back.